model/src/main/java/com/epam/esm/model/service/impl/GiftCertificateServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+package com.epam.esm.model.service.impl;
+
+import com.epam.esm.model.dao.GiftCertificateDao;
+import com.epam.esm.model.dao.entity.GiftCertificate;
+import com.epam.esm.model.dao.entity.SortType;
+import com.epam.esm.model.dao.exception.DaoException;
+import com.epam.esm.model.service.GiftCertificateService;
+import com.epam.esm.model.service.converter.impl.CertificateConverter;
+import com.epam.esm.model.service.dto.CertificateDTO;
+import com.epam.esm.model.service.exception.ServiceException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class GiftCertificateServiceImpl implements GiftCertificateService {
+
+    private final static Logger logger = Logger.getLogger(GiftCertificateServiceImpl.class);
+    private final GiftCertificateDao certificateDao;
+    private final CertificateConverter certificateConverter;
```

Comment:
Try to give new classes the same names, you should decide to use prefix Gift or not